File: src/main/java/com/EduTrack/EDUTRACK_Portfolio_MS/model/Trainee.java
package com.EduTrack.EDUTRACK_Portfolio_MS.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Trainee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String traineeId;
    private String firstName;
    private String lastName;
    private LocalDate dob;
    private String gender;
    private String maritalStatus;
    private String email;
    private String phoneNbr;
    private String address;
}

File: frontend/src/app/core/models/trainee.ts
export interface Trainee {
    id?: number;
    firstName: string;
    lastName: string;
    address: string;
    phoneNbr: string;
    traineeId: string;
    email: string;
    dob: Date;
    maritalStatus: string;
    gender: string;
    canEqual?: boolean;
    hashCode?: boolean;
    toString?: boolean;
}

export interface TraineeBuilder {
    id: number;
    firstName: string;
    lastName: string;
    email: string;
    phoneNbr: string;
    address: string;
    gender: string;
    dob: Date;
    maritalStatus: string;
    traineeId: string;
}